cpp/example_code/s3/put_website_config.cpp

```diff
@@ -1,103 +1,111 @@
+ */
 
-// snippet-start:[s3.cpp.put_website_config.code]
-bool AwsDoc::S3::PutWebsiteConfig(const Aws::String& bucketName, 
-    const Aws::String& indexPage, const Aws::String& errorPage, 
-    const Aws::String& region)
-{
-    Aws::Client::ClientConfiguration config;
+//! Routine which demonstrates configuring a website for an S3 bucket.
+/*!
+  \sa PutWebsiteConfig()
+  \param bucketName name of S3 bucket.
+  \param indexPage name of index page.
+  \param errorPage name of error page.
+  \param clientConfig Aws client configuration.
+*/
 
-    if (!region.empty())
-    {
-        config.region = region;
-
-    }
+// snippet-start:[s3.cpp.put_website_config.code]
+bool AwsDoc::S3::PutWebsiteConfig(const Aws::String &bucketName,
+                                  const Aws::String &indexPage, const Aws::String &errorPage,
+                                  const Aws::Client::ClientConfiguration &clientConfig) {
+    Aws::S3::S3Client client(clientConfig);
 
-    Aws::S3::S3Client s3_client(config);
-        
-    Aws::S3::Model::IndexDocument index_doc;
-    index_doc.SetSuffix(indexPage);
+    Aws::S3::Model::IndexDocument indexDocument;
+    indexDocument.SetSuffix(indexPage);
 
-    Aws::S3::Model::ErrorDocument error_doc;
-    error_doc.SetKey(errorPage);
+    Aws::S3::Model::ErrorDocument errorDocument;
+    errorDocument.SetKey(errorPage);
 
-    Aws::S3::Model::WebsiteConfiguration website_config;
-    website_config.SetIndexDocument(index_doc);
-    website_config.SetErrorDocument(error_doc);
+    Aws::S3::Model::WebsiteConfiguration websiteConfiguration;
+    websiteConfiguration.SetIndexDocument(indexDocument);
+    websiteConfiguration.SetErrorDocument(errorDocument);
 
     Aws::S3::Model::PutBucketWebsiteRequest request;
     request.SetBucket(bucketName);
-    request.SetWebsiteConfiguration(website_config);
+    request.SetWebsiteConfiguration(websiteConfiguration);
 
-    Aws::S3::Model::PutBucketWebsiteOutcome outcome = 
-        s3_client.PutBucketWebsite(request);
+    Aws::S3::Model::PutBucketWebsiteOutcome outcome =
+            client.PutBucketWebsite(request);
 
-    if (outcome.IsSuccess())
-    {
-        std::cout << "Success: Set website configuration for bucket '" 
-            << bucketName << "'." << std::endl;
-
-        return true;
+    if (!outcome.IsSuccess()) {
+        std::cerr << "Error: PutBucketWebsite: "
+                  << outcome.GetError().GetMessage() << std::endl;
     }
-    else
-    {
-        std::cout << "Error: PutBucketWebsite: "
-            << outcome.GetError().GetMessage() << std::endl;
-
-        return false;
+    else {
+        std::cout << "Success: Set website configuration for bucket '"
+                  << bucketName << "'." << std::endl;
     }
-    
-    return 1;
+
+    return outcome.IsSuccess();
 }
+// snippet-end:[s3.cpp.put_website_config.code]
 
-int main()
-{
+/*
+ *
+ *  main function
+ *
+ * Prerequisites: One S3 bucket containing an index page and an error page.
```

Comment:
* Prerequisites: Create one S3 bucket that contains an index page and an error page.